jboss-marshalling/src/main/java/org/infinispan/jboss/marshalling/commons/GenericJBossMarshaller.java

```diff
@@ -15,35 +13,31 @@
  */
 public final class GenericJBossMarshaller extends AbstractJBossMarshaller {
 
+   private final ClassLoader classLoader;
```

Comment:
Not sure about about keeping a hard ref to the ClassLoader. I've seen DefaultContextClassResolver using a WeakRef for some reason. Maybe @danberindei remembers why. Anyway, this is simple to avoid. Rather than giving a lengthy description, here what I have in mind:
```java
public final class GenericJBossMarshaller extends AbstractJBossMarshaller {

   public GenericJBossMarshaller() {
      this(null, null);
   }

   public GenericJBossMarshaller(ClassLoader classLoader) {
      this(classLoader, null);
   }

   public GenericJBossMarshaller(ClassWhiteList classWhiteList) {
      this(null, classWhiteList);
   }

   public GenericJBossMarshaller(ClassLoader classLoader, ClassWhiteList classWhiteList) {
      super();
      if (classLoader == null) {
         classLoader = getClass().getClassLoader();
      }
      baseCfg.setClassResolver(classWhiteList == null ?
            new DefaultContextClassResolver(classLoader) :
            new CheckedClassResolver(classWhiteList, classLoader)
      );
   }

   @Override
   public void initialize(ClassWhiteList classWhiteList) {
      ClassLoader classLoader = ((DefaultContextClassResolver) baseCfg.getClassResolver()).getClassLoader();
      baseCfg.setClassResolver(new CheckedClassResolver(classWhiteList, classLoader));
   }
}
```

Reply:
Done